### PR TITLE
Fixing Lidar for UE5 and be able to save its data to a .ply file using SensorSpawnerActor.

### PR DESCRIPTION
**Pull request by JoseM98**
#### Description

- The ParallelLineTrace method has been added to the engine, so it has been used in ShootLaser(...) method.
- Chaos ReadLock is done instead of PhysX ReadLock in SimulateLidar(...) method.
- Added the option to save .ply files with the PointCloud of the Lidar.
- To be able to access the PointCloud outside SimulateLidar(...) method, the points have been stored in an FVector. As they are stored in a CarlaLib class that has no getters.

Fixes #  

#### Where has this been tested?

  * **Platform(s):** Windows and Linux.
  * **Python version(s):** 3.8.10
  * **Unreal Engine version(s):** UE 5.3

#### Possible Drawbacks

- This change has to be added with the engine change, so that the ParallelLineTraceSingleByChannel(...) method does not give an error when compiling the ParallelLineTraceSingleByChannel(...) method.

### CODE REVIEW

File: Unreal/CarlaUE4/Plugins/Carla/Source/Carla/Sensor/RayCastLidar.cpp [hunk 1]
```diff
@@ -150,16 +150,34 @@ ARayCastLidar::FDetection ARayCastLidar::ComputeDetection(const FHitResult& HitI
...
     LidarData.ResetMemory(PointsPerChannel);
+    PointCloudResetMemory();
 
     for (auto idxChannel = 0u; idxChannel < Description.Channels; ++idxChannel) {
       for (auto& hit : RecordedHits[idxChannel]) {
         FDetection Detection = ComputeDetection(hit, SensorTransform);
         if (PostprocessDetection(Detection))
+        {
           LidarData.WritePointSync(Detection);
+          PointCloudWritePointSync(Detection);
```
Comment (Blyron) [inline Unreal/CarlaUE4/Plugins/Carla/Source/Carla/Sensor/RayCastLidar.cpp:161]: Would be nice if this operations is surrounded with an if condition and #if WITH_EDITOR

File: Unreal/CarlaUE4/Plugins/Carla/Source/Carla/Sensor/RayCastLidar.cpp [hunk 2]
```diff
@@ -150,16 +150,34 @@ ARayCastLidar::FDetection ARayCastLidar::ComputeDetection(const FHitResult& HitI
       PointsPerChannel[idxChannel] = RecordedHits[idxChannel].size();
 
     LidarData.ResetMemory(PointsPerChannel);
+    PointCloudResetMemory();
```
Comment (Blyron) [inline Unreal/CarlaUE4/Plugins/Carla/Source/Carla/Sensor/RayCastLidar.cpp:153]: Would be nice if this operations is surrounded with an if condition and #if WITH_EDITOR

File: Unreal/CarlaUE4/Plugins/Carla/Source/Carla/Sensor/RayCastLidar.cpp (hunk 1 above)
Comment (JoseM98) [reply Unreal/CarlaUE4/Plugins/Carla/Source/Carla/Sensor/RayCastLidar.cpp]: Fixed.

File: Unreal/CarlaUE4/Plugins/Carla/Source/Carla/Sensor/RayCastLidar.cpp (hunk 2 above)
Comment (JoseM98) [reply Unreal/CarlaUE4/Plugins/Carla/Source/Carla/Sensor/RayCastLidar.cpp]: Fixed.

**Review by Blyron — APPROVED** [review]

**Pull request merged**